### ISSUE TITLE
unexpected response: '220 smtp.gmail.com ESMTP i7sm2984410igt.18 - gsmtp'

### ISSUE THREAD

**Issue by bendiy** [Kozea/Multicorn#104]
I'm trying to test the IMAP FDW following the example:
http://multicorn.org/foreign-data-wrappers/#idimap-foreign-data-wrapper

I'm getting this error in pgAdmin and pqsl:
```
ERROR:  Error in python: abort
DETAIL:  unexpected response: '220 smtp.gmail.com ESMTP p82sm3981275ioi.14 - gsmtp'
```

I have tried disabling 2FA and some other settings in Gmail, but it doesn't seem to help.

I tried the same with a non-Gmail IMAP server and the FDW is working fine.

**Comment by rdunklau** [comment]
It appears you are connecting to an SMTP server. Could you try to set the "port" option to 993 (IMAPS) ?

By the way, testing your use case made me realize that there may be a bug with imap fdw and python 3, I'll look into it.

Thank you for your interest in Multicorn !

**Comment by bendiy** [comment]
@rdunklau You're right, I was following the example directly here:
http://multicorn.org/#idusage

It has port `465`, which is SMTP:
```
create foreign table gmail (
    "Message-ID" character varying,
    "From" character varying,
    "Subject" character varying,
    "payload" character varying,
    "flags" character varying[],
    "To" character varying) server multicorn_imap options (
        host 'imap.gmail.com',
        port '465',
        payload_column 'payload',
        flags_column 'flags',
        ssl 'True',
        login 'mylogin',
        password 'mypassword'
);
```

I fixed that and now I'm getting:
```
ERROR:  Error in python: error
DETAIL:  [ALERT] Please log in via your web browser: https://support.google.com/mail/accounts/answer/78754 (Failure)
```

I tried a different Gmail account and was able to get it working correctly. I think the above error is because the `multicorn.imapfdw.ImapFdw` isn't handling passwords with special characters correctly.

**Comment by rdunklau** [comment]
I tested it with an account with "special characters" in the password, and did not have any trouble. Gmail can be a pain to work with using standard protocols. Did you try what is advised here: https://support.google.com/accounts/answer/6010255 ?

Thank you for the report on the website, it is obviously an error on it, although I swear gmail was listening for IMAP connection on the SMTP port at the time this was written :). It is now fixed.
